### ISSUE TITLE
(90% sure) atmos bugged out and put plasma into distro.

### ISSUE THREAD

**Issue by meow20** [ParadiseSS13/Paradise#26963]
### Exploit Reports

- [X] I confirm this issue is not an exploit. (Required)

### BYOND Version

idk

### Issue Description

title. I hope it wasnt reported elsewhere, I can't find it if so.

### What did you expect to happen?

For gasses to go where humans direct them with pipes.

### What happened instead?

Half station burned down after plasma was put into distro, somehow.

### Why is this bad/What are the consequences?

I had to give heart attack to 3 players >:C

### Steps to reproduce the issue.

No fucking idea, quite a lot of atmos was done that round before the bug, during and after.

### When did the problem start happening?

today.

### Extra information

I am 90% sure its a bug and no human intentionally or by accident put plasma in distro, people did work on atmos, but none of their actions should have caused plasma to be put into distro.

- only plasma was pumped in, if it was grief or such, you'd think they'd pump in oxygen
- all toxin sources outside of atmos were untouched
- i investigated atmos, the way pipes and pumps were changed shouldn't put any plasma in distro, and it kept happening after 3 players involved messed around with it further ( i used VV to remove plasma from distro, it refilled itself 5~ times after i removed all of it until it was all gone))

i hope the runtimes are related and will help. I know the upper runtime is known about, the bottom one has not been mentioned on our public discord.

### Relevant log output/runtime error

```DM
[2024-10-02T08:21:19] Runtime in code/modules/atmospherics/machinery/datum_pipeline.dm:129: Cannot execute null.addMember().
   proc name: addMember (/obj/machinery/atmospherics/proc/addMember)
   usr: August Ringer (CKEY) (/mob/living/carbon/human)
   usr.loc: The reinforced floor (120,67,2) (/turf/simulated/floor/engine)
   src: the gas filter (/obj/machinery/atmospherics/trinary/filter)
   src.loc: the reinforced floor (120,67,2) (/turf/simulated/floor/engine)
   call stack:
   the gas filter (/obj/machinery/atmospherics/trinary/filter): addMember
   the pipe (/obj/machinery/atmospherics/pipe/simple): initialize_atmos_network
   the pipe (/obj/machinery/atmospherics/pipe/simple): on_construction
   the bent pipe fitting (/obj/item/pipe): wrench_act
   the rapid pipe dispenser (/obj/item/rpd): create_atmos_pipe
   the reinforced floor (120,68,2) (/turf/simulated/floor/engine): rpd_act
   the rapid pipe dispenser (/obj/item/rpd): afterattack
   the rapid pipe dispenser (/obj/item/rpd): melee_attack_chain
   August Ringer (/mob/living/carbon/human): ClickOn
   the reinforced floor (120,68,2) (/turf/simulated/floor/engine): Click
   CKEY (/client): Click


[2024-10-02T08:21:22] Runtime in code/modules/atmospherics/gasmixtures/gas_mixture.dm:669: Cannot read null.volume
   proc name: share many airs (/proc/share_many_airs)
   src: null
   call stack:
   share many airs(/list (/list))
   /datum/pipeline (/datum/pipeline): reconcile_air
   /datum/pipeline (/datum/pipeline): process
   Atmospherics (/datum/controller/subsystem/air): process_pipenets
   Atmospherics (/datum/controller/subsystem/air): fire_sleepless
   Atmospherics (/datum/controller/subsystem/air): fire
   Atmospherics (/datum/controller/subsystem/air): ignite
   Master (/datum/controller/master): RunQueue
   Master (/datum/controller/master): Loop
   Master (/datum/controller/master): StartProcessing
```

**Comment by silverplatedelta** [comment]
Assuming the runtime was thrown from the source, it sounds like a distribution pipe was connected via a filter. Due to the way we handle pipe layers now, its possible for these devices to connect to pipes, even if they are hidden under the floor, resulting in what seems to be teleporting of gas.

Would you be willing to share what map this occurred on, and share what modifications if any were made to the atmospheric system?

**Comment by meow20** [comment]
Cyberiad.
The modifications made were custom sm setup, turbine and bunch of other stuff so nothing helpful here.

**Comment by silverplatedelta** [comment]
Odd. Reason I asked is that things like thermomachines filters, and mixers can send gas though distro pipes if they run on top. If there was a mixer (as I suspect from the runtime) misplaced in atmos, it could cause this issue.

**Comment by Fordoxia** [comment]
Insight: when making custom SMs for Farragus, I have noticed it is possible to accidentally link distro to the location that I place the cold loop (secondary cooling is usually filled by a source of plasma) because of underfloor air pipes.

I forget the exact configuration, I need to get around to documenting it at some point.

**Comment by S34NW** [comment]
Yeah, there's a reason we don't run pipes over each other where possible when mapping. There is some undiagnosed issue causing pipelines to leak into other pipelines that cross/share the same tile
